experiments/modelfree/plot_noisy_observations.py

```diff
@@ -0,0 +1,141 @@
+    plt.figure(figsize=(10, 7))
+    sns.lmplot(data=transformed_df, x='log_noise', y='agent0_win_perc', hue='agent1_path',
+               logistic=logistic)
+    plt.title("{}: Noisy Zoo{} Observations vs Normal Zoos".format(subset_specs['env'],
+                                                                   subset_specs['agent0_path']))
+    if savefile is not None:
+        plt.savefig(savefile)
+    else:
+        plt.show()
+    plt.close()
+
+
+def generate_plots(experiment):
+    num_episodes = int(experiment.split("_")[1])
+
+    baseline_transformations = [
+        {'new_col': 'log_noise', 'old_col': 'noise_magnitude',
+         'func': lambda x: math.log(x)},
+        {'new_col': 'agent0_win_perc', 'old_col': 'agent0_wins',
+         'func': lambda x: x / num_episodes},
+        {'new_col': 'agent1_win_perc', 'old_col': 'agent1_wins',
+         'func': lambda x: x / num_episodes}
+    ]
+
+    zoo_path = os.path.join(RESULT_DIR, experiment, "noisy_zoo_observations.json")
+    adversary_path = os.path.join(RESULT_DIR, experiment, "noisy_adversary_observations.json")
+    with open(adversary_path, "r") as fp:
+        noisy_obs_against_adv = json.load(fp)
+
+    with open(zoo_path, "r") as fp:
+        noisy_obs_against_zoo = json.load(fp)
+    noisy_zoo_obs_df = pd.DataFrame(
+        [process_element_into_flat_dict(el) for el in noisy_obs_against_zoo])
+    noisy_adv_obs_df = pd.DataFrame(
+        [process_element_into_flat_dict(el) for el in noisy_obs_against_adv])
+    experiment_out_dir = os.path.join(OUT_DIR, experiment)
+
+    if not os.path.exists(experiment_out_dir):
+        os.mkdir(experiment_out_dir)
+    for short_env in env_lookup:
+        for zoo_id in range(1, available_zoos[short_env] + 1):
+            subset_params = {
+                'agent0_path': str(zoo_id),
+                'env': env_lookup[short_env]}
+
+            zoo_plot_path = os.path.join(experiment_out_dir,
+                                         f"{experiment}_ZooBaseline_"
+                                         f"{short_env}_AgainstZoo{zoo_id}")
+
+            adversary_plot_path = os.path.join(experiment_out_dir,
+                                               f"{experiment}_AdversaryTrained_"
+                                               f"{short_env}_AgainstZoo{zoo_id}")
+            noisy_multiple_opponent_subset_plot(noisy_zoo_obs_df, subset_specs=subset_params,
+                                                transform_specs=baseline_transformations,
+                                                savefile=zoo_plot_path)
+            noisy_adversary_opponent_subset_plot(noisy_adv_obs_df, subset_specs=subset_params,
+                                                 transform_specs=baseline_transformations,
+                                                 savefile=os.path.join(adversary_plot_path))
+
+
+if __name__ == "__main__":
+    for experiment_run in ["ep_500_5-22_single_zoo", "ep_100_5-21", "ep_500_5-22_all_zoo"]:
```

Comment:
Will the path names always be these?

Reply:
These are the current experiments I have run; this is just an example usage, and which experiments you want to visualize will be a function of which you have most recently run, which I don't know how I would anticipate

Reply:
Okay, sure. The reason I was putting it in main was that I think it's useful to have an example of "here are directories you can use to have a version this code that is known to work", but I can just remove that if it doesn't seem useful.